smarts/core/remote_agent_buffer.py

```diff
@@ -44,6 +49,12 @@ def __init__(self, zoo_worker_addrs=None, buffer_size=3):
         self._local_zoo_worker = None
         self._local_zoo_worker_addr = None
 
+        assert isinstance(
+            auth_key, (str, type(None))
+        ), f"Received auth_key of type {type(auth_key)}, but need auth_key of type <class 'string'> or <class 'NoneType'>."
+        self._auth_key = auth_key if auth_key else ""
+        self._auth_key_conn = str.encode(auth_key) if auth_key else None
```

Comment:
nit: This is fine but I am not convinced we need to store this variable since it is mostly a duplication of `self._auth_key` and `str.encode()` is not costly.